game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/general/FiringGroupSplitterGeneralTest.java

```diff
@@ -86,13 +88,60 @@ void oneWaitingToDieFiringUnitVsOneTargetableUnitMakesOneFiringGroup() {
   }
 
   @Test
-  void firingUnitsAreExcludedAccordingToThePredicate() {
+  void offensiveNormalTypeExcludesFirstStrike() {
     final Unit targetUnit = givenAnyUnit();
     final Unit fireUnit = givenAnyUnit();
+    final Unit fireUnit2 = givenUnitFirstStrike();
+
+    final List<FiringGroup> firingGroups =
+        FiringGroupSplitterGeneral.of(OFFENSE, FiringGroupSplitterGeneral.Type.OFFENSIVE_NORMAL, "")
+            .apply(
+                givenBattleStateBuilder()
+                    .gameData(givenGameData().withAlliedAirIndependent(true).build())
+                    .attacker(attacker)
+                    .defender(defender)
+                    .attackingUnits(List.of(fireUnit, fireUnit2))
+                    .defendingUnits(List.of(targetUnit))
+                    .build());
+
+    assertThat(firingGroups, hasSize(1));
+    assertThat(firingGroups.get(0).getFiringUnits(), contains(fireUnit));
+  }
+
+  @Test
+  void defensiveNormalTypeExcludesFirstStrike() {
+    final Unit targetUnit = givenAnyUnit();
+    final Unit fireUnit = givenAnyUnit();
+    final Unit fireUnit2 = givenUnitFirstStrike();
+
+    final List<FiringGroup> firingGroups =
+        FiringGroupSplitterGeneral.of(OFFENSE, FiringGroupSplitterGeneral.Type.DEFENSIVE_NORMAL, "")
```

Comment:
The empty string arg seems very similar to a 'null' arg or a boolean parameter. IN general those kinds of a args are not self-explanatory and make it difficult for someone to read and use an API. Notably, you can't just use the API, you have to look up which values are allowed to then realize empty string is the way to go. One would wonder, "can I use null here"? In any case, it makes you look up the implementation which is a heuristic for a difficult to use API. Similar for those reading, they'll be unsure why an empty string is needed.

Generally the fix is to have method overloads and have an API that omits the unnecessary parameter.